utils/hct/HashStability.py

```diff
@@ -0,0 +1,158 @@
+    proc.communicate()
+    return proc.wait()
+
+
+def run_hash_stablity_test(args, dxc_path, dxa_path, test_name, working_dir):
+    empty_env = os.environ.copy()
+    # clear PATH to make sure dxil.dll are not found.
+    empty_env["PATH"] = ""
+    args[0] = dxc_path
+    # run original compile
+    proc = subprocess.Popen(args, cwd=working_dir,
+                           # env=empty_env,
+                            # don't writ output to stdout
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=USE_CLOSE_FDS)
+    proc.communicate()
+    res = proc.wait()
+    if res != 0:
+        # original compilation failed, don't run hash test.
+        return True, "Original compilation failed."
+
+    # run normal compile
+    normal_out = os.path.join(working_dir, 'Output', test_name+'.normal.out')
+    res = normal_compile(args, normal_out, working_dir, empty_env)
+    if res != 0:
+        # strip_reflect failed, return fail.
+        return False, "Adding Qstrip_reflect failed compilation."
+
+    normal_hash = extract_hash(dxa_path, normal_out, working_dir, empty_env)
+    if normal_hash is None:
+        return False, "Fail to get hash for normal compilation."
+
+    # run debug compilation
+    debug_out = os.path.join(working_dir, 'Output', test_name+'.dbg.out')
+    res = debug_compile(args, debug_out, working_dir, empty_env)
+    if res != 0:
+        # strip_reflect failed, return fail.
+        return False, "Adding Qstrip_reflect and Zi failed compilation."
+
+    debug_hash = extract_hash(dxa_path, debug_out, working_dir, empty_env)
+    if debug_hash is None:
+        return False, "Fail to get hash for normal compilation."
+
+    # compare normal_hash and debug_hash.
+    if filecmp.cmp(normal_hash, debug_hash):
+        # hash match, return pass.
+        return True, "Hash match."
+    else:
+        # hash mismatch
+        return False, "Hash mismatch."
+
+################################################
+################################################
+# For running from the command-line
+
+def usage():
+    print("""HashStability.py - Run hash stability test.
+
+Usage:
+HashStability.py <dxc_args> - args to run dxc like " "%dxc -ECSMain -Tcs_6_0 D:\test.hlsl"
+    <bin_dir>       - path to dxc.exe and dxa.exe
+""")
+    return 1
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Description of your program')
+    parser.add_argument('-a','--argument', help='origin command line options to run dxc like \"dxc -ECSMain -Tcs_6_0 D:\\test.hlsl\"', required=True)
+    parser.add_argument('-p','--path', help='path to find dxc and dxa', required=True)
+    args = vars(parser.parse_args())
+    dxc_args = args['argument'].split(' ')
+    dxc_args[0] = "%dxc"
+    bin_dir = args['path']
+    dxc_path = os.path.join(bin_dir, 'dxc.exe')
+    dxa_path = os.path.join(bin_dir, 'dxa.exe')
+    working_dir = os.getcwd()
```

Comment:
Aren't these getting passed in? I don't know why they are constructed twice

Reply:
Simplify the required parameter to make it easier to run thru command line.